benchmarks/offload.py

```diff
@@ -47,31 +48,39 @@ def train(args: argparse.Namespace):
     transform = ToTensor()
     dataloader = DataLoader(
         FakeData(image_size=(1, args.inputs, args.inputs), num_classes=args.outputs, transform=transform),
-        batch_size=BATCH,
+        batch_size=args.batch_size,
     )
 
-    def train_epoch():
+    def train_epoch(args):
         model.train()
+        iter_count = 2
         for batch_inputs, batch_outputs in dataloader:
             batch_inputs, batch_outputs = batch_inputs.to("cuda"), batch_outputs.to("cuda")
-
             start = time.time_ns()
-            optimizer.zero_grad()
-            inputs = batch_inputs.reshape(-1, args.inputs * args.inputs)
-            output = model(inputs)
-            loss = criterion(output, target=batch_outputs)
-            loss.backward()
-            optimizer.step()
-
-            print("Loss {:.2f} - throughput {:.2f}fps".format(loss.item(), BATCH / (time.time_ns() - start) * 10 ** 9))
-
-    train_epoch()
+            with torch.autograd.profiler.profile(use_cuda=True, profile_memory=True) as prof:
+                optimizer.zero_grad()
+                inputs = batch_inputs.reshape(-1, args.inputs * args.inputs)
+                with torch.autograd.profiler.record_function("model_training"):
+                    output = model(inputs)
+                    loss = criterion(output, target=batch_outputs)
+                    loss.backward()
+                    optimizer.step()
+            prof.export_chrome_trace("/tmp/mpi_prof")
```

Comment:
could that be made a little more generic ? I'm guessing that it's superseded by the other PR actually ?

Reply:
It is. Does the change in the next PR address this?